### PR TITLE
Fix validating id token encryption algorithm and method

### PR DESCRIPTION
**Pull request by Sathiyakugan**
### Proposed changes in this pull request
Fix https://github.com/wso2/product-is/issues/7773

Added validation methods to check the Id Token and the Algorithms exists

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 1]
```diff
@@ -1295,6 +1308,44 @@ void logOnInvalidConfig(String appName, String tokenType, long defaultValue) {
...
+
+    /**
+     * Get the IdToken Encryption Algorithm registered by the user and filter the allowed one.
+     *
+     * @param application Application user have registered
+     * @return idTokenEncryptionAlgorithm
+     * @throws IdentityOAuthAdminException Identity OAuthAdmin exception.
+     */
+    String filterIdTokenEncryptionAlgorithm(OAuthConsumerAppDTO application) throws IdentityOAuthAdminException {
+        List<String> supportedIdTokenEncryptionAlgorithms = OAuthServerConfiguration.getInstance()
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java]: Add new line

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 2]
```diff
@@ -1295,6 +1308,44 @@ void logOnInvalidConfig(String appName, String tokenType, long defaultValue) {
...
+                .getSupportedIdTokenEncryptionAlgorithm();
+        String idTokenEncryptionAlgorithm = application.getIdTokenEncryptionAlgorithm();
+        if (!supportedIdTokenEncryptionAlgorithms.contains(idTokenEncryptionAlgorithm)) {
+            String msg = String.format("'%s' IdToken Encryption Method is not allowed.", idTokenEncryptionAlgorithm);
+            throw handleClientError(INVALID_REQUEST, msg);
+        }
+        return idTokenEncryptionAlgorithm;
+    }
+
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java]: Remove extra line

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 3]
```diff
@@ -1295,6 +1308,44 @@ void logOnInvalidConfig(String appName, String tokenType, long defaultValue) {
         return requestedScopeValidators;
     }
 
+
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java]: remove extra line

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java (hunk 1 above)
Comment (Sathiyakugan) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java]: Fixed

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java (hunk 2 above)
Comment (Sathiyakugan) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java]: Fixed

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java (hunk 3 above)
Comment (Sathiyakugan) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java]: Fixed

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 4]
```diff
@@ -279,9 +279,16 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
...
                     app.setRequestObjectSignatureValidationEnabled(
                             application.isRequestObjectSignatureValidationEnabled());
-                    app.setIdTokenEncryptionEnabled(application.isIdTokenEncryptionEnabled());
-                    app.setIdTokenEncryptionAlgorithm(application.getIdTokenEncryptionAlgorithm());
-                    app.setIdTokenEncryptionMethod(application.getIdTokenEncryptionMethod());
+
+                    // Validate IdToken Encryption configurations.
+                    if (application.isIdTokenEncryptionEnabled()) {
+                        app.setIdTokenEncryptionEnabled(application.isIdTokenEncryptionEnabled());
+                        if (application.isIdTokenEncryptionEnabled()) {
```
Comment (darshanasbg) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:286]: Isn't this redundant?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 5]
```diff
@@ -449,9 +456,15 @@ public void updateConsumerApplication(OAuthConsumerAppDTO consumerAppDTO) throws
             oauthappdo.setScopeValidators(filterScopeValidators(consumerAppDTO));
             oauthappdo.setRequestObjectSignatureValidationEnabled(consumerAppDTO
                     .isRequestObjectSignatureValidationEnabled());
-            oauthappdo.setIdTokenEncryptionEnabled(consumerAppDTO.isIdTokenEncryptionEnabled());
-            oauthappdo.setIdTokenEncryptionAlgorithm(consumerAppDTO.getIdTokenEncryptionAlgorithm());
-            oauthappdo.setIdTokenEncryptionMethod(consumerAppDTO.getIdTokenEncryptionMethod());
+            // Validate IdToken Encryption configurations.
+            if (consumerAppDTO.isIdTokenEncryptionEnabled()) {
+                oauthappdo.setIdTokenEncryptionEnabled(consumerAppDTO.isIdTokenEncryptionEnabled());
+                if (consumerAppDTO.isIdTokenEncryptionEnabled()) {
```
Comment (darshanasbg) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:462]: Isn't this redundant?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 6]
```diff
@@ -1295,6 +1308,45 @@ void logOnInvalidConfig(String appName, String tokenType, long defaultValue) {
...
 
+
+    /**
+     * Get the IdToken Encryption Method registered by the user and filter the allowed one.
+     *
+     * @param application Application user have registered
+     * @return idTokenEncryptionMethod
+     * @throws IdentityOAuthAdminException Identity OAuthAdmin exception.
+     */
+    String filterIdTokenEncryptionMethod(OAuthConsumerAppDTO application) throws IdentityOAuthAdminException {
```
Comment (darshanasbg) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:1319]: Should this be a private method?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java (hunk 4 above)
Comment (Sathiyakugan) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:286]: Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/commit/4adb2395c29dd91bf7b6d835c2bb878f96f3741e#diff-1e769d65ab3dba0fc2f6ab9b801f1e72

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java (hunk 5 above)
Comment (Sathiyakugan) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:462]: Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/commit/4adb2395c29dd91bf7b6d835c2bb878f96f3741e#diff-1e769d65ab3dba0fc2f6ab9b801f1e72

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java (hunk 6 above)
Comment (Sathiyakugan) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:1319]: Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1470

**Pull request merged**